sdk/dapp-kit/src/slices/walletSlice.ts

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import type { Wallet, WalletAccount, WalletWithRequiredFeatures } from '@mysten/wallet-standard';
+import type { StateCreator } from 'zustand';
+
+export type WalletActions = {
+	setAccountSwitched: (selectedAccount: WalletAccount) => void;
+	setWalletConnected: (
+		wallet: WalletWithRequiredFeatures,
+		selectedAccount: WalletAccount | null,
+	) => void;
+	setWalletDisconnected: () => void;
+	setWalletRegistered: (updatedWallets: WalletWithRequiredFeatures[]) => void;
+	setWalletUnregistered: (
+		updatedWallets: WalletWithRequiredFeatures[],
+		unregisteredWallet: Wallet,
+	) => void;
+};
+
+export type WalletSlice = {
+	wallets: WalletWithRequiredFeatures[];
+	currentAccount: WalletAccount | null;
+	lastConnectedAccountAddress: string | null;
+} & WalletActions &
+	(
```

Comment:
I'm not sure if we want the `connectionStatus` field at all, if we want disconnected/connecting/connected as possible options, or just disconnected/connected 🤔 

My arguing point for all three was that you might want this data outside the context of connecting/disconnecting in your application even though it isn't a popular use case

Reply:
Yeah I go back and forth, I'm fine with either tbh, it's more that I'm not sure what the value is in practice.